pupil-spa/src/app/virtual-keyboard/virtual-keyboard.component.ts

```diff
@@ -0,0 +1,36 @@
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'app-virtual-keyboard',
+  templateUrl: './virtual-keyboard.component.html',
+  styleUrls: ['./virtual-keyboard.component.scss']
+})
+export class VirtualKeyboardComponent implements OnInit {
+
+  constructor() { }
+
+  answer = '';
+  questionNum = 1;
+  factor1 = 7;
+  factor2 = 2;
+  total = 14;
+  num = 2;
+
+  onClickAnswer(number) {
+    this.answer = `${this.answer}${number}`;
```

Comment:
@pris54 there was a bug around login. It's been fixed and merged to your branch now